### ISSUE TITLE
quic-fuzzer crash

### ISSUE THREAD

**Issue by kroeckx** [openssl/openssl#22538]
This file: [crash-8a9b91d8c01d945afb73140a53845f3c96b59d06.gz](https://github.com/openssl/openssl/files/13188845/crash-8a9b91d8c01d945afb73140a53845f3c96b59d06.gz)

Returns:
```
quic-client: ssl/quic/quic_ackm.c:1304: int ossl_ackm_on_timeout(OSSL_ACKM *): Assertion `lost_pkts != NULL' failed.
==3842704== ERROR: libFuzzer: deadly signal
    #0 0x5558541bda74 in __sanitizer_print_stack_trace (quic-client+0x2b2a74) (BuildId: 881a33dde8a7787c0541a3cf95a7b4da8e331a27)
    #1 0x5558541f2a19 in fuzzer::PrintStackTrace() (quic-client+0x2e7a19) (BuildId: 881a33dde8a7787c0541a3cf95a7b4da8e331a27)
    #2 0x5558541d757d in fuzzer::Fuzzer::CrashCallback() (quic-client+0x2cc57d) (BuildId: 881a33dde8a7787c0541a3cf95a7b4da8e331a27)
    #3 0x7f5e5102550f  (/lib/x86_64-linux-gnu/libc.so.6+0x3c50f) (BuildId: 8a1bf172e710f8ca0c1576912c057b45f90d90d8)
    #4 0x7f5e510730fb in __pthread_kill_implementation nptl/pthread_kill.c:43:17
    #5 0x7f5e51025471 in raise signal/../sysdeps/posix/raise.c:26:13
    #6 0x7f5e5100f4b1 in abort stdlib/abort.c:79:7
    #7 0x7f5e5100f3d4 in __assert_fail_base assert/assert.c:92:3
    #8 0x7f5e5101e3a1 in __assert_fail assert/assert.c:101:3
    #9 0x5558542d1a1c in ossl_ackm_on_timeout ssl/quic/quic_ackm.c:1304:9
    #10 0x5558542ab97e in ch_tick ssl/quic/quic_channel.c:1947:13
    #11 0x55585423c41f in ossl_quic_reactor_tick ssl/quic/quic_reactor.c:109:5
    #12 0x5558542386c7 in quic_read ssl/quic/quic_impl.c:2701:9
    #13 0x55585420aa0a in SSL_read ssl/ssl_lib.c:2342:11
    #14 0x5558541bf774 in FuzzerTestOneInput fuzz/quic-client.c:123:23
    #15 0x5558541d8dbc in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) (quic-client+0x2cddbc) (BuildId: 881a33dde8a7787c0541a3cf95a7b4da8e331a27)
    #16 0x5558541d8299 in fuzzer::Fuzzer::RunOne(unsigned char const*, unsigned long, bool, fuzzer::InputInfo*, bool, bool*) (quic-client+0x2cd299) (BuildId: 881a33dde8a7787c0541a3cf95a7b4da8e331a27)
    #17 0x5558541d9896 in fuzzer::Fuzzer::MutateAndTestOne() (quic-client+0x2ce896) (BuildId: 881a33dde8a7787c0541a3cf95a7b4da8e331a27)
    #18 0x5558541da1c5 in fuzzer::Fuzzer::Loop(std::vector<fuzzer::SizedFile, std::allocator<fuzzer::SizedFile>>&) (quic-client+0x2cf1c5) (BuildId: 881a33dde8a7787c0541a3cf95a7b4da8e331a27)
    #19 0x5558541c5b6a in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) (quic-client+0x2bab6a) (BuildId: 881a33dde8a7787c0541a3cf95a7b4da8e331a27)
    #20 0x5558541bfb41 in main (quic-client+0x2b4b41) (BuildId: 881a33dde8a7787c0541a3cf95a7b4da8e331a27)
    #21 0x7f5e510106c9 in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16
    #22 0x7f5e51010784 in __libc_start_main csu/../csu/libc-start.c:360:3
    #23 0x555854195200 in _start (quic-client+0x28a200) (BuildId: 881a33dde8a7787c0541a3cf95a7b4da8e331a27)
```

**Comment by kroeckx** [comment]
That at least looks similar to https://github.com/openssl/openssl/pull/22368#issuecomment-1763494447 that we had problems reproducing.

**Comment by mattcaswell** [comment]
Ah - I wondered when that would appear again!

**Comment by mattcaswell** [comment]
My proposed fix in #22541. The ack manager loss detection timer was firing because a packet was not acked within the time period that we calculated it should be acked within when it was first sent. However when the timer fires the rtt info has changed since when we sent the packet and it is no longer considered lost yet. We seem to be assuming that if the loss detection timer fires then there must be lost packets.
